x-pack/plugins/uptime/public/components/common/header/page_header.tsx

```diff
@@ -36,40 +38,32 @@ const StyledPicker = styled(EuiFlexItem)`
   }
 `;
 
-export const PageHeader = () => {
-  const isCertRoute = useRouteMatch(CERTIFICATES_ROUTE);
-  const isSettingsRoute = useRouteMatch(SETTINGS_ROUTE);
-  const isStepDetailRoute = useRouteMatch(STEP_DETAIL_ROUTE);
-
-  const DatePickerComponent = () =>
-    isCertRoute ? (
-      <CertRefreshBtn />
-    ) : (
-      <StyledPicker grow={false} style={{ flexBasis: 485 }}>
-        <UptimeDatePicker />
-      </StyledPicker>
-    );
-
-  const isMonRoute = useRouteMatch(MONITOR_ROUTE);
-
-  if (isStepDetailRoute) {
-    return null;
-  }
-
+export const PageHeader = ({
+  includeSpacer = false,
+  showCertificateRefreshBtn = false,
+  showDatePicker = false,
+  showMonitorTitle = false,
+  showTabs = false,
+}: Props) => {
   return (
     <>
       <SyntheticsCallout />
       <EuiFlexGroup alignItems="center" justifyContent="spaceBetween" wrap responsive={false}>
         <EuiFlexItem>
-          <PageTabs />
+          {showMonitorTitle && <MonitorPageTitle />}
+          {showTabs && <PageTabs />}
         </EuiFlexItem>
         <EuiFlexItem grow={false}>
           <ToggleAlertFlyoutButton />
         </EuiFlexItem>
-        {!isSettingsRoute && <DatePickerComponent />}
+        {showCertificateRefreshBtn && <CertRefreshBtn />}
+        {showDatePicker && (
+          <StyledPicker grow={false} style={{ flexBasis: 485 }}>
+            <UptimeDatePicker />
+          </StyledPicker>
+        )}
       </EuiFlexGroup>
-      {isMonRoute && <EuiHorizontalRule margin="m" />}
-      {!isMonRoute && <EuiSpacer size="m" />}
```

Comment:
i believe this includeSpacer prop isn't needed, can we move this perhaps to top of those pages? where it's needed?